whitepaper.tex

```diff
@@ -765,115 +809,146 @@
   \href{https://docs.sel4.systems/projects/sel4/status.html}{seL4
     project status page} for details.
 
-  \SSect{The CAmkES component framework}\label{s:camkes}
+  \SSect{The seL4 Microkit}\label{s:microkit}
+
+  The Microkit is a thin abstraction layer over seL4 that hides much
+  of seL4's complexities at the cost of imposing the restriction of a
+  static system architecture: At system configuration time, all
+  modules and their authorised communication channels are known.
 
   \begin{figure}[b]
     \centering
-    \includegraphics[width=0.6\textwidth]{camkes}
-    \caption{CAmkES components and connectors.}
-    \label{f:camkes}
+    \includegraphics[width=0.6\textwidth]{microkit}
+    \caption{Microkit abstractions.}
+    \label{f:microkit}
   \end{figure}
 
-  \href{https://trustworthy.systems/projects/TS/camkes/trustcomp.pml}{CAmkES}
-  is a component framework that allows you to reason about a
-  system architecturally, i.e.\ as a collection of sandboxed
-  components with defined communication channels. \autoref{f:camkes}
-  shows the main abstractions.
-  \begin{description}
-  \item[Components] are represented as square boxes. They represent
-     programs, code and data, encapsulated by seL4.
-  \item[Interfaces] are shown as decorations on the components. They
-    define how a component can be invoked, or can invoke others. An
-    interface is either imported (invoking an interface of another
-    component) or exported (able to be invoked by another component's
-    imported interface), except for the shared-memory interface, which
-    is symmetric.
-  \item[Connectors] connect like interfaces by linking an importing with an
-    exporting interface. Connectors in CAmkES are always one-to-one,
-    but broadcast or multicast functionality can be implemented on top
-    of this model by building components that copy inputs to multiple outputs.
-  \end{description}
+  \autoref{f:microkit} shows these abstractions (yes, that's all!) The
+  \emph{protection domain} (PD) is the process abstraction: it
+  represents a program in execution in its own, isolated address
+  space. Two PDs can be connected by a \emph{communication channel}
+  and may also share \emph{memory regions} that are mapped in one or
+  more PDs' address spaces. A PD may be a VM (running a guest OS), but
+  other PDs are oblivious to that fact.
```

Comment:
I would prefer to say something along that lines of "A VM is similar to a PD in that it is an isolated component with the additional capability of running a guest OS". The way it's presented to users of Microkit it is a different concept even though conceptually it's basically a PD.